src/optional-functions/billing/usage-poller/handler.ts

```diff
@@ -0,0 +1,102 @@
+import { GetValueCommand } from "@stedi/sdk-client-stash";
+import { randomUUID } from "crypto";
+import fetch, { RequestInit } from "node-fetch";
+import { stashClient } from "../../../lib/clients/stash.js";
+import { PARTNERS_KEYSPACE_NAME } from "../../../lib/constants.js";
+import {
+  ProcessDeliveriesInput,
+  processDeliveries,
+} from "../../../lib/deliveryManager.js";
+import { requiredEnvVar } from "../../../lib/environment.js";
+import { ConfigurationSchema } from "./Configuration.js";
+import { configurationKey } from "./constants.js";
+
+interface UsageExceededPayload {
+  message: string;
+  threshold: number;
+  currentUsage: number;
+  estimatedUsage: number;
+}
+
+export const handler = async (): Promise<undefined | UsageExceededPayload> => {
+  const configurationResponse = await stashClient().send(
+    new GetValueCommand({
+      keyspaceName: PARTNERS_KEYSPACE_NAME,
+      key: configurationKey,
+    })
+  );
+
+  if (!configurationResponse.value) {
+    console.log("no configuration for usage poller");
+    return;
+  }
+
+  const configuration = ConfigurationSchema.parse(configurationResponse.value);
+
+  const now = new Date();
+  const isoNow = now.toISOString();
+  const usageUrl = `https://api.billing.stedi.com/2021-09-01/usage?period=${encodeURIComponent(
+    isoNow
+  )}`;
+
+  const usageResponse = await fetch(usageUrl, {
+    headers: {
+      Authorization: `Key ${requiredEnvVar("STEDI_API_KEY")}`,
+      Accept: "application/json",
+    },
+  } satisfies RequestInit);
+
+  const { subtotal: usage } = (await usageResponse.json()) as {
+    subtotal: number;
+  };
+
+  const monthStart = new Date(now.getFullYear(), now.getMonth(), 1, 0, 0, 0);
+  const nextMonthStart = new Date(
+    now.getFullYear(),
+    now.getMonth() + 1,
+    1,
+    0,
+    0,
+    0
+  );
+
+  const pctTimeElapsedInMonth = Math.max(
+    0.03,
```

Comment:
Why round up to a full day on the first day, but not others (I'm making the assumption that getTime() returns something like seconds since epoch)?